tracer/src/Datadog.Trace.Coverage.collector/AssemblyInfo.cs

```diff
@@ -13,6 +13,6 @@
 [assembly: InternalsVisibleTo("Datadog.Trace.Tools.Runner.Tests, PublicKey=002400000480000094000000060200000024000052534131000400000100010025b855c8bc41b1d47e777fc247392999ca6f553cdb030fac8e3bd010171ded9982540d988553935f44f7dd58cb4b17fbb92653d5c2dc5112696886665b317c6f92795bf64beab2405c501c8a30cb1b31b1541ed66e27d9823169ec2815b00ceeeecc8d5a1bf43db67d2961a3e9bea1397f043ec07491709649252f5565b756c5")]
 
 // Sanity check for _OR_GREATER compiler directives
-#if !NETCOREAPP3_1_OR_GREATER && !NET461_OR_GREATER && !NETSTANDARD2_0
+#if !NETCOREAPP3_1_OR_GREATER && !NET461_OR_GREATER && !NETCOREAPP2_1_OR_GREATER
```

Comment:
I think `NETCOREAPP3_1_OR_GREATER` is redundant if `NETCOREAPP2_1_OR_GREATER` is defined?
```suggestion
#if !NET461_OR_GREATER && !NETCOREAPP2_1_OR_GREATER
```

Reply:
... or check specifically for `NETCOREAPP2_1` like you did in `Datadog.Trace`
```suggestion
#if !NETCOREAPP3_1_OR_GREATER && !NET461_OR_GREATER && !#if !NETCOREAPP2_1
```

Reply:
TBH, I suspect these checks are all a bit redundant tbh - they were added because some SDK versions in .NET 6 didn't support `OR_GREATER`, but we're way past that now, so not sure they add much value? WDYT? 🤔

Reply:
I trust your judgement here.

Reply:
Updated to `NET461_OR_GREATER && !NETCOREAPP2_1_OR_GREATER` seeing as this is specifically about checking the `OR_GREATER` checks are available, rather than mandating specific TFMs

Reply:
Done 👍 https://github.com/DataDog/dd-trace-dotnet/pull/5260/commits/8362a6f134756d23c3596a3e4c6904d59145cb88